mypy/newsemanal/semanal.py

```diff
@@ -2223,57 +2237,45 @@ def analyze_lvalue(self, lval: Lvalue, nested: bool = False,
             self.fail('Invalid assignment target', lval)
 
     def analyze_name_lvalue(self,
-                            lval: NameExpr,
+                            lvalue: NameExpr,
                             explicit_type: bool,
                             is_final: bool) -> None:
         """Analyze an lvalue that targets a name expression.
 
         Arguments are similar to "analyze_lvalue".
         """
-        if self.is_alias_for_final_name(lval.name):
+        if lvalue.node:
+            # This has been bound already in a previous iteration.
+            return
```

Comment:
Why do we do this for names but not for attributes? Maybe leave a comment/TODO about this.

Reply:
I'll create an issue about dealing with attribute definitions (through `self.x`).